---
title: Phylogeographic analysis of spatial diffusion
author: Alejandro Manuel Ferreiro
date: '2022-05-03'
slug: phylogeographic-analysis-of-spatial-diffusion
categories:
  - Genetics
  - Phylogeography
tags:
  - BEAST
subtitle: ''
summary: ''
authors: []
lastmod: '2022-05-03T09:57:13-03:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The analyzes implemented in phylogeographic studies have traditionally allowed inferring the genetic structure and demographic processes that have shaped the current distribution of genetic lineages. Many times these demographic changes have been implicitly associated with range expansions of the species. However, these analyzes do not explicitly test these hypotheses. To do this, in this post we are going to introduce ourselves to the use of the SPREAD application. This software allows the explicit reconstruction of the distribution of lineages in space and time, taking into account the phylogenetic uncertainty (posterior probability of the nodes), and the spatial diffusion process (Kms in Ma).</p>
<p>This post is an adaptation of <a href="http://beast.community/workshop_continuous_diffusion_yfv">BEAST tutorial</a>.</p>
</div>
<div id="software-needed" class="section level1">
<h1>Software needed</h1>
<ol style="list-style-type: decimal">
<li>BEAUTi</li>
<li>BEAST v. 1.10.4</li>
<li>TRACER v. 1.7.2</li>
<li>Tree Annotator</li>
<li>spreaD3</li>
</ol>
</div>
<div id="how-do-i-perform-spatial-diffusion-analysis" class="section level1">
<h1>How do I perform spatial diffusion analysis?</h1>
<div id="starting-files" class="section level2">
<h2>Starting files</h2>
<p>We start with a <a href="./docs/sequences.fas">fasta file</a> that contains the sequences, and a <a href="./docs/coordinates.nex">nexus file</a> containing geographical coordinates of each sequence. AdemásIn addition, it is necessary to know the evolutionary model that best fits the sequence matrix, which can be calculated in software such as <a href="https://github.com/ddarriba/jmodeltest2">jModelTest</a>.</p>
<p>In addition to these files, in order to geographically represent and visualise the spatial dynamics of the species, a vector geographic file of the type geojson will be needed. This can be obtained in R using the following code:</p>
<pre class="r"><code>library(sf)</code></pre>
<pre><code>## Linking to GEOS 3.9.1, GDAL 3.4.3, PROJ 7.2.1; sf_use_s2() is TRUE</code></pre>
<pre class="r"><code>library(rnaturalearth)

# Download polygons of South American country boundaries:
Sudam = ne_states(iso_a2 = c(&quot;AR&quot;, &quot;BO&quot;, &quot;PY&quot;, &quot;BR&quot;, &quot;CL&quot;, &quot;UY&quot;, &quot;PE&quot;, &quot;EC&quot;, &quot;VE&quot;, &quot;CO&quot;, &quot;GY&quot;, &quot;SR&quot;, &quot;FR&quot;),
                                     returnclass = &quot;sf&quot;)
write_sf(obj = Sudam, dsn = &quot;./ProvSudam.geojson&quot;)</code></pre>
<pre><code>## Warning in CPL_write_ogr(obj, dsn, layer, driver,
## as.character(dataset_options), : GDAL Error 6: DeleteLayer() not supported by
## this dataset.</code></pre>
</div>
<div id="beauti-v.-1.10.4" class="section level2">
<h2>BEAUTi v. 1.10.4</h2>
<p>In this programme we are going to prepare our data to generate an xml file, which can be read by the BEAST programme. To do this::
1. Import the file <code>sequence.fas</code>. Once imported, check in the <code>Partitions</code> tab that the data is correct.
2. In the <code>Traits</code> tab, add a new data partition with geographical coordinates. Click <code>Import traits</code> to import <code>coordinates.nex</code>. Then, click <code>Create partition from trait...</code> and name the partition as you wish, for example <code>location</code>.
3. <code>Sites</code> tab.
4. <code>Clocks</code> tab.
5. <code>Trees</code> tab.
6. <code>States</code> tab.
7. <code>Priors</code> tab.
8. <code>MCMC</code> tab.</p>
</div>
<div id="beast-v-1.10.4" class="section level2">
<h2>BEAST v 1.10.4</h2>
<p>You can run the analysis in the software or BEAST, or you can use an internet server like <a href="https://www.phylo.org/">CIPRES</a> to save work for our machine.</p>
</div>
<div id="tracer-1.7.2" class="section level2">
<h2>Tracer 1.7.2</h2>
<p>After the run, we must check that the analysis has had a chain length long enough to reach ESS values greater than 200 (convergence). For this we open the <code>.log</code> file generated in the Tracer program.
If those of ESS have posterior probabilities greater than 200, we can check the <strong>diffusion rate</strong> which is found to be <code>location.diffusionRate</code>. This is indicated in km/year. In species, where data on dispersal rates are available, this value is important for comparison with previous studies.</p>
</div>
<div id="treeannotator-v.-1.10.4" class="section level2">
<h2>TreeAnnotator v. 1.10.4</h2>
<p>This software allows to generate trees that summarise the information of the 10000 trees retrieved by Bayesian inference. In this software you must define the <code>Burn-in</code> of low probability trees. Usually 12-25% of the trees are discarded or “burned” (usually aiming for a total of 10,000 trees before burning).
Leave other options by default, and load the file generated after BEAST run with the extention <code>.trees</code>. As output you can generate a file <code>.tree</code> that can be read in <a href="http://tree.bio.ed.ac.uk/software/figtree/">Figtree</a>.</p>
</div>
<div id="spread3" class="section level2">
<h2>spreaD3</h2>
<p>Finally, to generate the spatial patterns of haplotypes and their ancestors, the spreaD3 software must be opened.
* <code>Data</code> tab. Load the .tree file obtained with TreeAnontator. Next, select <code>location1</code> as latitude and <code>location2</code> as longitude. After that, you must load a geojson file with the baseline map you will use, for eg. South American bordes. Click <code>Generate JSON</code> and name the json file of output, including the extention .json.
* <code>Rendering</code> tab. Load the json file generated in the previous tab. Render it in kml form, defining the name followed by the extension.
* This will generate a .kml file in the folder which can then be opened using google earth.</p>
</div>
</div>
